Webapp employee

rayolanderos opened the pull request:
All employee methods (add, delete, update, get all)
Employee dashboard layout

Missing:
Input validations and cleaning (particularily for phone number formatting)

rayolanderos committed:
Initial methods for employee

Model, page, and layout for employees

rayolanderos committed:
Employee methods

All employee methods (add, delete, update, get all)
Employee dashboard layout

Missing:
Input validations and cleaning (particularily for phone number formatting)

rayolanderos committed:
Merge branch 'master' into webapp-employee

rayolanderos committed:
Methods for order, user

Models: Order, OrderBeer, User
Methods:
- get order
- get all orders
- add order
- add user
- get user
- update user

Layout for Orders

rayolanderos committed:
Final details for orders

Orders get details of relations and displays in layout.

rayolanderos committed:
Merge branch 'webapp-orders' into webapp-employee

rayolanderos commented:
Added Order and User methods into this pull request. 
Order:
- Get all
- Get single
- Add order

User:
- Get single
- Add User
- Update User


There is also a screen in the webapp that displays all orders with details for better control

gleono commented on webapp/controllers/employees.py:
```diff
@@ -0,0 +1,87 @@
+            api_uri = self.uri_for('add-employee', _full=True)
+        elif (action == "updateEmployee"):
+            api_uri = self.uri_for('update-employee', _full=True)
+            logging.info("++++++++++++ updating employee_data +++++++++++++++");
```
Do you wanna keep all the logging?

rayolanderos replied:
no

gleono commented on webapp/models/order_beer.py:
```diff
@@ -0,0 +1,12 @@
+class OrderBeer(ndb.Model):
+    
+    order_id = ndb.IntegerProperty()
+    beer_id = ndb.IntegerProperty()
```
I think these 2 should be `ndb.KeyProperty()` to actually have references of the objects

gleono commented on webapp/services/add_order.py:
```diff
@@ -0,0 +1,59 @@
+
+			for order_beer in order_beers:
+				order_beer = OrderBeer(
+					order_id = int( order_id ),
```
You would use the keys directly here, instead of integers.

gleono commented on webapp/services/add_user.py:
```diff
@@ -0,0 +1,42 @@
+
+		else:
+			res = { "msg" : "That user already exists in the system or something went wrong. Please try again.", "success": False }
+			self.response.out.write(json.dumps(res))
```
a post request normally doesn't write a response. Is standard to use status code instead.

gleono commented on webapp/services/get_user.py:
```diff
@@ -0,0 +1,36 @@
+		user = User.get_by_id(user_id)
+
+		if user != None:
+			user_creation_date = user.creation_date
```
You can do it in one line, no need for extra variables. :)
`user_creation_date = user.creation_date.strftime('%m/%d/%Y')`

gleono commented on webapp/services/update_employee.py:
```diff
@@ -0,0 +1,35 @@
+			employee.put()
+
+			res = { "msg" : "Employee successfully updated", "success": True, "employee_id" : employee_id }
+			self.response.out.write(json.dumps(res))
```
Same here for status codes.

gleono commented on webapp/services/update_user.py:
```diff
@@ -0,0 +1,33 @@
+			user.put()
+
+			res = { "msg" : "User successfully updated", "success": True, "user_id" : user_id }
+			self.response.out.write(json.dumps(res))
```
and here

rayolanderos replied:
Using all of these responses as output messages in the layout for error / success alerts. Is there a reason why you want me to update all of this? I am aware of REST protocol but even for ajax a post response in json format is super handy.

gleono commented on webapp/www/layout.html:
```diff
@@ -27,7 +29,7 @@
       <ul class="navbar-nav navbar-sidenav" id="exampleAccordion">
         <li class="nav-item" data-toggle="tooltip" data-placement="right" title="Tap List">
-          <a class="nav-link" href="/tap-list">
+          <a class="nav-link" href="/tap_list">
```
Thank you!

gleono reviewed (changes requested):
Please address the key properties in the OrderBeer model and the status code for post methods in the services. Other than that looks great 👍. Good work.

rayolanderos committed:
Merging Orders and user methods

Removed logging, updated Description to accomodate longer text

rayolanderos commented:
Ended up leaving keys and post request responses for error msging. 

Removed all useless logging and updated the beer description property to allow for longer text.
